newm-server/src/main/kotlin/io/newm/server/features/marketplace/daemon/MarketplaceMonitorDaemon.kt

```diff
@@ -0,0 +1,96 @@
+package io.newm.server.features.marketplace.daemon
+
+import io.ktor.server.application.ApplicationEnvironment
+import io.newm.chain.grpc.MonitorAddressResponse
+import io.newm.server.features.cardano.repo.CardanoRepository
+import io.newm.server.features.marketplace.repo.MarketplaceRepository
+import io.newm.server.ktx.getSecureConfigString
+import io.newm.shared.daemon.Daemon
+import io.newm.shared.koin.inject
+import io.newm.shared.ktx.error
+import io.newm.shared.ktx.getConfigBoolean
+import io.newm.shared.ktx.getConfigLong
+import io.newm.shared.ktx.info
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.apache.curator.framework.recipes.leader.LeaderLatch
+import org.apache.curator.framework.recipes.leader.LeaderLatchListener
+import org.koin.core.parameter.parametersOf
+import org.slf4j.Logger
+import kotlin.time.Duration.Companion.seconds
+
+private const val LEADER_LATCH_PATH = "/leader-latches/marketplace-monitor"
```

Comment:
What are the instances where you would want something like this inside a companion object vs just outside of the class?

Reply:
What I've seen (at least in Android land), is that in the early days of  Kotlin everyone was using Companion objects. But it seems like in the more recent times the tendency is to declare constants outside.